src/python/pants/backend/python/rules/inject_init.py

```diff
@@ -0,0 +1,42 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from pants.backend.python.subsystems.pex_build_util import identify_missing_init_files
+from pants.engine.fs import EMPTY_DIRECTORY_DIGEST, Digest, Snapshot
+from pants.engine.isolated_process import ExecuteProcessRequest, ExecuteProcessResult
+from pants.engine.rules import rule
+from pants.engine.selectors import Get
+from pants.util.objects import datatype
+
+
+class InitInjectedDigest(datatype([('directory_digest', Digest)])): pass
+
+
+@rule(InitInjectedDigest, [Snapshot])
+def inject_init(snapshot):
+  """Ensure that every package has an __init__.py file in it."""
+  missing_init_files = tuple(sorted(identify_missing_init_files(snapshot.files)))
+  if not missing_init_files:
+    new_init_files_digest = EMPTY_DIRECTORY_DIGEST
+  else:
+    # TODO(7718): add a builtin rule for FilesContent->Snapshot.
+    touch_init_request = ExecuteProcessRequest(
+      argv=("/usr/bin/touch",) + missing_init_files,
```

Comment:
There is a very near term plan to fix that, so I think that it is fine in this case.

Reply:
It's not clear what the significance of the proposed builtin rule is to the line below the comment. If it could include something about not having absolute paths to files in this comment, I would be satisfied. I don't see why that's something we don't want to do here.